modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkPluginExtensionModelParser.java

```diff
@@ -0,0 +1,126 @@
+import static org.mule.runtime.module.extension.mule.internal.dsl.MuleSdkDslConstants.MULE_SDK_EXTENSION_PREFIX_PARAMETER_NAME;
+import static org.mule.runtime.module.extension.mule.internal.dsl.MuleSdkDslConstants.MULE_SDK_EXTENSION_REQUIRED_ENTITLEMENT_PARAMETER_NAME;
+import static org.mule.runtime.module.extension.mule.internal.dsl.MuleSdkDslConstants.MULE_SDK_EXTENSION_REQUIRES_ENTERPRISE_LICENSE_PARAMETER_NAME;
+import static org.mule.runtime.module.extension.mule.internal.dsl.MuleSdkDslConstants.MULE_SDK_EXTENSION_VENDOR_PARAMETER_NAME;
+
+import org.mule.metadata.api.TypeLoader;
+import org.mule.runtime.api.meta.Category;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.internal.model.ExtensionModelHelper;
+import org.mule.runtime.module.extension.internal.loader.java.property.LicenseModelProperty;
+import org.mule.runtime.module.extension.internal.loader.parser.ExtensionModelParser;
+import org.mule.runtime.module.extension.internal.loader.parser.XmlDslConfiguration;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * {@link ExtensionModelParser} implementation for Mule SDK plugins
+ *
+ * @since 4.5.0
+ */
+public class MuleSdkPluginExtensionModelParser extends AbstractMuleSdkExtensionModelParser {
+
+  private String name;
+  private Category category;
+  private String vendor;
+  private String namespace;
+  private Optional<XmlDslConfiguration> xmlDslConfiguration;
+  private LicenseModelProperty licenseModelProperty;
+
+  public MuleSdkPluginExtensionModelParser(ArtifactAst ast, TypeLoader typeLoader, ExtensionModelHelper extensionModelHelper) {
+    super(ast, typeLoader, extensionModelHelper);
+    parseMetadata(getExtensionComponentAst(ast));
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public Category getCategory() {
+    return category;
+  }
+
+  @Override
+  public String getVendor() {
+    return vendor;
+  }
+
+  @Override
+  public Optional<XmlDslConfiguration> getXmlDslConfiguration() {
+    return xmlDslConfiguration;
+  }
+
+  @Override
+  public String getNamespace() {
+    return namespace;
+  }
+
+  @Override
+  public LicenseModelProperty getLicenseModelProperty() {
+    return licenseModelProperty;
+  }
+
+  @Override
+  protected Stream<ComponentAst> getTopLevelElements(ArtifactAst ast) {
+    return getExtensionComponentAst(ast).directChildrenStream();
+  }
+
+  private ComponentAst getExtensionComponentAst(ArtifactAst ast) {
+    // At this point we can assume there is only one top level component which is the extension:extension component
+    // We don't need to check for this because it should be guaranteed by previous validations
+    return ast.topLevelComponents().get(0);
+  }
+
+  private void parseMetadata(ComponentAst extensionComponentAst) {
```

Comment:
rename to `parseStructure`, not only for consistency but because this isn't parsing metadata.

Reply:
Renamed